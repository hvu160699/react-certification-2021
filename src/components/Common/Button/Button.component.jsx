import styled from '@emotion/styled';

const Button = styled.button`
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.5rem;
  color: rgba(156, 163, 175, 1);
  background-color: transparent;
  border-width: 0px;
  border-radius: 0.375rem;
  outline: unset;
  transition: all 250ms ease-in-out;
  cursor: pointer;

  &:hover {
    background-color: #e5e7eb;
  }
`;

export default Button;
