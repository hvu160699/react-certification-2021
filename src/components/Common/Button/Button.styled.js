import styled from '@emotion/styled';

const StyledButton = styled.button`
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.375rem 0.75rem;
  border-width: 0px;
  border-radius: 0.375rem;
  outline: unset;
  transition: all 250ms ease-in-out;
  cursor: pointer;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
`;

const PrimaryButton = styled(StyledButton)`
  color: rgba(30, 58, 138, 1);
  background-color: rgba(219, 234, 254, 1);

  &:hover {
    background-color: rgba(191, 219, 254, 1);
  }
`;

const LightButton = styled(StyledButton)`
  color: #212529;
  background-color: #f8f9fa;

  &:hover {
    background-color: #e2e6ea;
  }
`;

const Button = { PrimaryButton, LightButton };

export default Button;
