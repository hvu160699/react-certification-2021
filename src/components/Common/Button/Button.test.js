import React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Button from './Button.component';

describe('Button', () => {
  it('Should render a button with any text', () => {
    const { getByText } = render(<Button>My button</Button>);
    expect(getByText(/.+/s)).toBeInTheDocument();
  });

  it('Shoul render a button that can contain multiply element', () => {
    const button = renderer
      .create(
        <Button>
          <img src={null} alt="img" />
          <span>Img button</span>
        </Button>
      )
      .toJSON();

    expect(button).toMatchSnapshot();
  });

  it('Should triggers function when the button is clicked', () => {
    const onClickFn = jest.fn();
    render(<Button onClick={onClickFn}>Click me</Button>);

    fireEvent.click(screen.getByText(/Click me/i));
    expect(onClickFn).toHaveBeenCalledTimes(1);
  });
});
