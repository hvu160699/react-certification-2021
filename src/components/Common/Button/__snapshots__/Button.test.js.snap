// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Shoul render a button that can contain multiply element 1`] = `
<button
  className="css-oaqt4x"
>
  <img
    alt="img"
    src={null}
  />
  <span>
    Img button
  </span>
</button>
`;
